### PR TITLE
Call to onMetadata handles exceptions

### PR DESCRIPTION
**Pull request by ruicraveiro**
This PR solves the issue https://github.com/grpc/grpc-dart/issues/341, where the server will crash when an exception is thrown from the onMetadata method.

It handles the exception in order for the exception to be returned to the client, the request not being processed, yet the server not crashing.

Fixes https://github.com/grpc/grpc-dart/issues/341

### CODE REVIEW

**Comment by mraleph** [comment]
This looks good to me. Thank you for the contribution. Please add a test and rebase. After that I will approve and merge.

**Comment by ruicraveiro** [comment]
Hi, please let me know if the test is OK. Thanks!

File: test/round_trip_test.dart
Comment (mraleph) [inline test/round_trip_test.dart:101]: Consider

```dart
await expectLater(testClient.stream(1).toList(), throwsA(isA<GrpcError>()));
```

Because the way the test is currently written does not actually verify that the error is being thrown, it just check that if it is thrown then it would be `GrpcError`.

**Review by mraleph — CHANGES_REQUESTED** [review]
Almost LGTM. 

We are currently blocked on finishing the sync of internal fork with external repo, but once this completes this PR would ready to go.

**Comment by ruicraveiro** [comment]
I just submitted the change, but now Travis is failing with a totally unrelated problem on Windows. Anything I can do?

**Review by mraleph — APPROVED** [review]
LGTM.

**Comment by mraleph** [comment]
I have landed a workaround for the issue with Travis in dd34af2de42194c7604e61baff530c7fdd43f097

**Comment by ruicraveiro** [comment]
It works! :-)

**Comment by ruicraveiro** [comment]
Hi! There were a few commits on the master branch that required me to update this branch and now Windows builds are broken again, also with an unrelated issue to the commits on this branch.

**Comment by mraleph** [comment]
Windows builds should be fixed now.

**Comment by ruicraveiro** [comment]
3 greens! :-)

**Comment by mraleph** [comment]
@ruicraveiro FYI  I am trying to figure out the Windows issues (Travis on Windows has failed on the added tests) so I am making some changes to the PR myself. 

As soon as I figure it out I will merge.

**Comment by ruicraveiro** [comment]
@mraleph Thank you!

**Pull request merged**
